### PR TITLE
#11004: moreh: use env var for kernel src search path

### LINKED ISSUE #11004: moreh: use env var for kernel src search path
Create env var: TT_METAL_KERNEL_PATH
Should work like, eg, LD_LIBRARY_PATH
Kernel compilation will use this as search paths for the kernel bin
This is in additional to TT_METAL_HOME and absolute paths (supported today)

Due date: early September

### PR DESCRIPTION
**Pull request by sagarwalTT**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/11004)

### What's changed
Added an env variable named `TT_METAL_KERNEL_PATH` that allows kernel files located in this path to be used.

### Checklist
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/host_api.hpp [hunk 1]
```diff
@@ -114,12 +114,12 @@ Program CreateProgram();
...
+ * | Argument     | Description                                                                                                                                 | Type                                                     | Valid Range | Required |
+ * |--------------|---------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------|-------------|----------|
+ * | program      | The program to which this kernel will be added to                                                                                           | Program &                                                |             | Yes      |
+ * | file_name    | Path to kernel src. Assumed to be absolute/relative to CWD, but will fall back to relative path from TT_METAL_HOME/TT_METAL_KERNEL_PATH.    | const std::string &                                      |             | Yes      |
```
Comment (tt-aho) [inline tt_metal/host_api.hpp:120]: Should we be explicit on the order of path resolution?

File: tests/tt_metal/tt_metal/test_create_kernel_with_kernel_path_env_var.cpp [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #include <gtest/gtest.h>
```
Comment (pgkeller) [inline tests/tt_metal/tt_metal/test_create_kernel_with_kernel_path_env_var.cpp:5]: does this test run normally or under CI?  thinking we should update the CI script to create the dir under /tmp/metal_test and pass the env var in so it gets run.  I think this should be ok...

**Review by pgkeller — CHANGES_REQUESTED** [review]

File: tests/tt_metal/tt_metal/test_create_kernel_with_kernel_path_env_var.cpp (hunk 2 above)
Comment (sagarwalTT) [reply tests/tt_metal/tt_metal/test_create_kernel_with_kernel_path_env_var.cpp:5]: @tt-rkim Can I create a CI test for this in the `All Post-commit tests` workflow or should I add it to another workflow?

Comment (sagarwalTT) [reply tests/tt_metal/tt_metal/test_create_kernel_with_kernel_path_env_var.cpp]: Added to `All Post-commit tests` workflow: https://github.com/tenstorrent/tt-metal/actions/runs/11003253105/job/30552388812#step:6:23

**Review by tt-rkim — APPROVED** [review]

**Review by pgkeller — APPROVED** [review]

**Pull request merged**